x-pack/plugins/security_solution/public/management/components/endpoint_responder/lib/console_commands_definition.ts

```diff
@@ -421,5 +458,52 @@ export const getEndpointConsoleCommands = ({
     });
   }
 
+  // Kubernetes commands are currently behind feature flag
+  if (isKubernetesCommandsEnabled) {
+    consoleCommands.push({
+      name: 'kube-list',
+      about: getCommandAboutInfo({
+        aboutInfo: i18n.translate('xpack.securitySolution.endpointConsoleCommands.kubeList.about', {
+          defaultMessage: 'List Kubernetes Resources (pod or deployment)',
+        }),
+        isSupported: doesEndpointSupportCommand('kube-list'),
+      }),
+      RenderComponent: KubeListActionResult,
+      meta: {
+        endpointId: endpointAgentId,
+        capabilities: endpointCapabilities,
+        privileges: endpointPrivileges,
+      },
+      exampleUsage: 'kube-list --resource pod --comment "listing kubernetes pods"',
+      exampleInstruction: ENTER_RESOURCE_NAME,
+      validate: capabilitiesAndPrivilegesValidator,
+      mustHaveArgs: true,
+      args: {
+        comment: {
+          required: false,
+          allowMultiples: false,
+          about: COMMENT_ARG_ABOUT,
+        },
+        resource: {
+          required: true,
+          allowMultiples: false,
+          exclusiveOr: false,
+          about: i18n.translate(
+            'xpack.securitySolution.endpointConsoleCommands.kubeList.resource.arg.about',
+            {
+              defaultMessage: 'A kubernetes resource to list (pod or deployment)',
```

Comment:
Suggestion in order to clarify the valid values allowed:

```suggestion
              defaultMessage: 'A kubernetes resource to list. Valid value are "pod" or "deployment"',
```